acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/privacy/multitenancy/OnChainMultiTenancyAcceptanceTest.java

```diff
@@ -0,0 +1,697 @@
+        multiTenancyPrivacyNode.getTokenForTenant(tenant));
+    final EventEmitter eventEmitter =
+        privacyNode.execute(
+            privateContractTransactions.createSmartContractWithPrivacyGroupId(
+                EventEmitter.class,
+                privacyNode.getTransactionSigningKey(),
+                POW_CHAIN_ID,
+                tenant,
+                privacyGroupId));
+
+    final String transactionHash = getContractDeploymentCommitmentHash(eventEmitter);
+
+    // check that a member can get the transaction receipt
+    privacyNodeBesu.useAuthenticationTokenInHeaderForJsonRpc(
+        multiTenancyPrivacyNode.getTokenForTenant(tenant));
+    privacyNode.verify(
+        privateTransactionVerifier.validPrivateTransactionReceipt(
+            transactionHash,
+            (PrivateTransactionReceipt) eventEmitter.getTransactionReceipt().get()));
+    assertThat(
+            privacyNode
+                .execute(
+                    privacyTransactions.privGetCode(
+                        privacyGroupId,
+                        Address.fromHexString(eventEmitter.getContractAddress()),
+                        "latest"))
+                .toHexString())
+        .isEqualTo(eventEmitterDeployed);
+
+    // check that getting the transaction receipt does not work if you are not a member
+    privacyNodeBesu.useAuthenticationTokenInHeaderForJsonRpc(
+        multiTenancyPrivacyNode.getTokenForTenant(removedTenant));
+    privacyNode.verify(
+        privateTransactionVerifier.noPrivateTransactionReceipt(
+            transactionHash)); // // TODO: returning null because the RPC is using the enclave key
+
+    // check that getting the code of the event emitter does not work when you are not a member
+    assertThatThrownBy(
+            () ->
+                privacyNode.execute(
+                    privacyTransactions.privGetCode(
+                        privacyGroupId,
+                        Address.fromHexString(eventEmitter.getContractAddress()),
+                        "latest")))
+        .hasMessageContaining("Unauthorized");
+
+    final LogFilterJsonParameter filterParameter =
+        new LogFilterJsonParameter(
+            "earliest",
+            "latest",
+            List.of(eventEmitter.getContractAddress()),
+            Collections.emptyList(),
+            null);
+
+    // create a valid filter
+    privacyNodeBesu.useAuthenticationTokenInHeaderForJsonRpc(
+        multiTenancyPrivacyNode.getTokenForTenant(tenant));
+    final String filterId =
+        privacyNode.execute(privacyTransactions.newFilter(privacyGroupId, filterParameter));
+
+    //        THIS does not work because I cannot find out what the nonce is "Unable to determine
+    // nonce for account in group"
+    //        // check that sending a transaction does not work if you are not a member
+    //
+    // privacyNodeBesu.useAuthenticationTokenInHeaderForJsonRpc(multiTenancyPrivacyNode.getTokenForTenant(removedTenant));
```

Comment:
What are we gonna do about this? Does that mean that something is broken?

Reply:
Done! The way we are using web3j this cannot work